### ISSUE TITLE
Metro server crashes

### ISSUE THREAD

**Issue by smohamedjavid** [status-im/status-react#16341]
# Problem

The Metro server is crashing after 1-2 seconds after the server is started.

```log
➜  status-mobile ✗ make run-metro  
Configuring Nix shell for target 'clojure'...
No changes detected.
Done in: 1.159s
yarn run v1.22.19
$ react-native start
warn Package react-native-blob-util contains invalid configuration: "dependency.hooks" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
warn Package react-native-navigation contains invalid configuration: "dependency.assets" is not allowed,"dependency.hooks" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
                                                      
                        #######                       
                   ################                   
                #########     #########               
            #########             ##########          
        #########        ######        #########      
       ##########################################     
      #####      #####################       #####    
      #####          ##############          #####    
      #####    ###       ######       ###    #####    
      #####    #######            #######    #####    
      #####    ###########    ###########    #####    
      #####    ##########################    #####    
      #####    ##########################    #####    
      #####      ######################     ######    
       ######        #############        #######     
         #########        ####       #########        
              #########          #########            
                  ######### #########                 
                       #########                      
                                                      
                                                      
                    Welcome to Metro!
              Fast - Scalable - Integrated



To reload the app press "r"
To open developer menu press "d"

jest-haste-map: Watchman crawl failed. Retrying once with node crawler.
  Usually this happens when watchman isn't running. Create an empty `.watchmanconfig` file in your project's root folder or initialize a git or hg repository in your project.
  Error: Watchman error: A non-recoverable condition has triggered.  Watchman needs your help!
The triggering condition was at timestamp=1687286390: opendir(/Users/javid/Projects/status-mobile/node_modules/metro-core/node_modules/jest-regex-util/build) -> Too many open files
All requests will continue to fail with this message until you resolve
the underlying problem.  You will find more information on fixing this at
https://facebook.github.io/watchman/docs/troubleshooting.html#poison-opendir. Make sure watchman is running for this project. See https://facebook.github.io/watchman/docs/troubleshooting.
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: A non-recoverable condition has triggered.  Watchman needs your help!
The triggering condition was at timestamp=1687286390: opendir(/Users/javid/Projects/status-mobile/node_modules/metro-core/node_modules/jest-regex-util/build) -> Too many open files
All requests will continue to fail with this message until you resolve
the underlying problem.  You will find more information on fixing this at
https://facebook.github.io/watchman/docs/troubleshooting.html#poison-opendir

    at ChildProcess.<anonymous> (/Users/javid/Projects/status-mobile/node_modules/fb-watchman/index.js:207:21)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)
Emitted 'error' event on WatchmanWatcher instance at:
    at Client.<anonymous> (/Users/javid/Projects/status-mobile/node_modules/metro/node_modules/jest-haste-map/build/watchers/WatchmanWatcher.js:150:10)
    at Client.emit (node:events:513:28)
    at ChildProcess.<anonymous> (/Users/javid/Projects/status-mobile/node_modules/fb-watchman/index.js:214:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  watchmanResponse: {
    version: '2023.01.30.00',
    error: 'A non-recoverable condition has triggered.  Watchman needs your help!\n' +
      'The triggering condition was at timestamp=1687286390: opendir(/Users/javid/Projects/status-mobile/node_modules/metro-core/node_modules/jest-regex-util/build) -> Too many open files\n' +
      'All requests will continue to fail with this message until you resolve\n' +
      'the underlying problem.  You will find more information on fixing this at\n' +
      'https://facebook.github.io/watchman/docs/troubleshooting.html#poison-opendir\n'
  }
}

Node.js v18.16.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [run-metro] Error 1
```

#### Expected behaviour

The Metro server should work as expected without any errors.

### Additional Information

- Operating System: macOS 13.4
- Xcode version: `v14.3.1`
- Node: `v18.16.0`

**Comment by smohamedjavid** [comment]
The error says `Too many open files`. I tried killing the server with the `watchman shutdown-server` command in the Nix shell (targeting watchman) and restarted the system. No luck. I keep getting the error. Nix is on the latest version (`2.14.1`) and on the latest commit of `develop`.

I'm getting this error after the Nix changes are merged. It was working well before that.

**Comment by siddarthkay** [comment]
cc @yakimant @jakubgs 
Is it probably related to the very recent `nodejs` bump from `nixpkgs` upgrade? : https://github.com/status-im/status-mobile/commit/c97559793cd8af8d09ba38fc68f9f5b51545eb9c  
not sure because this doesn't happen for me locally

**Comment by yakimant** [comment]
Couldn't reproduce locally on macOS.
@smohamedjavid, can you post the following:
```
sysctl kern.maxfiles
sysctl kern.maxfilesperproc
ulimit -n
```

@siddarthkay, did you check locally on Linux? Can you try on macOS too?

**Comment by yakimant** [comment]
@smohamedjavid, please also post
```
lsof | wc -l
lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30
```
before/after `make run-metro`

**Comment by siddarthkay** [comment]
@yakimant : 
output from my mac : 
```log
siddarthkumar@Siddarths-MacBook-Pro status-mobile % sysctl kern.maxfiles
kern.maxfiles: 184320
siddarthkumar@Siddarths-MacBook-Pro status-mobile % sysctl kern.maxfilesperproc
kern.maxfilesperproc: 92160
siddarthkumar@Siddarths-MacBook-Pro status-mobile % ulimit -n
256
```

output from my linux : 
```log
siddarth@siddarth-ROG-Strix-GA35DX-G35DX:~/code/status-mobile$ sysctl kern.maxfiles
sysctl: cannot stat /proc/sys/kern/maxfiles: No such file or directory
siddarth@siddarth-ROG-Strix-GA35DX-G35DX:~/code/status-mobile$ sysctl kern.maxfilesperproc
sysctl: cannot stat /proc/sys/kern/maxfilesperproc: No such file or directory
siddarth@siddarth-ROG-Strix-GA35DX-G35DX:~/code/status-mobile$ ulimit -n
1024
```

no crash for me while running metro on both linux , macOS.. maybe its a rosetta thing?

**Comment by siddarthkay** [comment]
@yakimant : before running metro for me on MacOs: 
```log
siddarthkumar@Siddarths-MacBook-Pro status-mobile % lsof | wc -l
   86529
siddarthkumar@Siddarths-MacBook-Pro status-mobile % lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30

 571 MobileCal
 574 com.apple
 577 Collectio
 580 sharingd
 586 homed
 610 maild
 623 searchd
 624 siriinfer
 641 assistant
 696 com.apple
 698 webstorm
 713 NPKCompan
 713 PhotosFil
 715 StatusIm
 719 MobileSaf
 727 PhotosRel
 735 coreidvd
 737 callservi
 737 com.apple
 742 PhotosPos
 756 ScreenTim
 766 nanosyste
 771 locationd
 801 NewsToday
 854 PosterBoa
 902 Spotlight
 957 healthapp
1097 nim_statu
1114 SpringBoa
1986 com.apple

```

while metro is running : 
```log
siddarthkumar@Siddarths-MacBook-Pro status-mobile % lsof | wc -l

   86765
siddarthkumar@Siddarths-MacBook-Pro status-mobile % lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30

 571 MobileCal
 574 com.apple
 577 Collectio
 580 sharingd
 586 homed
 610 maild
 623 searchd
 624 siriinfer
 641 assistant
 696 com.apple
 698 webstorm
 713 NPKCompan
 713 PhotosFil
 717 StatusIm
 719 MobileSaf
 727 PhotosRel
 735 coreidvd
 737 callservi
 737 com.apple
 742 PhotosPos
 756 ScreenTim
 766 nanosyste
 771 locationd
 801 NewsToday
 854 PosterBoa
 902 Spotlight
 957 healthapp
1109 nim_statu
1114 SpringBoa
1986 com.apple
```

**Comment by yakimant** [comment]
Wow, `node_modules` is 71292 files and 11336 folders.
Is it wort adding it or some of it's subfolders to `ignore_dirs` in `.watchmanconfig`?

**Comment by jakubgs** [comment]
@siddarthkay it's possible I guess. But I cannot reproduce it locally. Works fine for me.

I guess @smohamedjavid could try to `git revert --no-commit c9755979` and see if that fixes it for him.

**Comment by mmilad75** [comment]
Same error happening for me

**Comment by rasom** [comment]
I experience same problem at the moment

**Comment by codemaster115** [comment]
Same issue here

**Comment by vkjr** [comment]
`ulimit -n` says I have 7000 files limit.

Tried:
- to set `ulimit -n 1000000` in the shell before running `make run-metro`
- to add `ulimit -n 1000000` before all commands inside `scripts/start-react-native.sh` and start `make run-metro`

Neither of these worked

**Comment by smohamedjavid** [comment]
I purged Nix to install it using Rosetta Terminal (to see if it's working). Now, I'm stuck with the `permission denied` error (https://github.com/status-im/status-mobile/issues/16344). 😢

**Comment by yakimant** [comment]
@smohamedjavid @mmilad75 @rasom @codemaster115 @vkjr, please share the following to help investigation:

1. `uname -s`
2. `sysctl kern.maxfiles`
3. `sysctl kern.maxfilesperproc`
4. `ulimit -n`
5. `lsof | wc -l`
6. `lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30`

@siddarthkay, @flexsurfer, what do you think of adding `node_modules` to `ignore_dirs` in `.watchmanconfig`? This might fix the issue and improve the overall performance as amount of files in this folder is huge.

**Comment by smohamedjavid** [comment]
```
➜  ~ uname -s
Darwin
➜  ~ sysctl kern.maxfiles
kern.maxfiles: 122880
➜  ~ sysctl kern.maxfilesperproc
kern.maxfilesperproc: 61440
➜  ~ ulimit -n
256
➜  ~ lsof | wc -l
   10437
➜  ~ lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30
  76 AppleIDSe
  76 CleanMyMa
  79 Grammarly
  82 suggestd
  83 Finder
  85 System
  87 com.apple
  97 SecurityP
 102 java
 103 Notificat
 104 Brave\x20
 106 figma_age
 109 Terminal
 129 CleanMyMa
 130 Brave\x20
 136 Discord
 137 Brave\x20
 139 Discord
 140 Spotlight
 149 Google
 178 Electron
 180 Brave\x20
 183 Brave\x20
 203 Google
 273 corespotl
 308 UserEvent
 333 Brave\x20
 373 java
 376 Code\x20-
 544 java

```

**Comment by codemaster115** [comment]
```
➜  status-mobile git:(develop) ✗ uname -s
Darwin
➜  status-mobile git:(develop) ✗ sysctl kern.maxfiles
kern.maxfiles: 122880
➜  status-mobile git:(develop) ✗ sysctl kern.maxfilesperproc
kern.maxfilesperproc: 61440
➜  status-mobile git:(develop) ✗ ulimit -n
256
➜  status-mobile git:(develop) ✗ lsof | wc -l
   14122
➜  status-mobile git:(develop) ✗ lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30
 101 Brave\x20
 101 CleanMyMa
 103 loginwind
 105 Arc\x20He
 112 Code\x20H
 127 CleanMyMa
 127 Code\x20H
 128 AppleSpel
 129 Notificat
 130 SetappAge
 133 krisp
 135 iTerm2
 150 Discord
 152 suggestd
 166 com.apple
 174 Discord
 189 Notes
 198 Electron
 212 Brave\x20
 217 OpenPhone
 233 corespotl
 251 Arc\x20He
 258 Raycast
 279 Brave\x20
 285 Mail
 291 UserEvent
 308 Telegram
 413 Arc
 580 Brave\x20
 698 Private```

**Comment by vkjr** [comment]
```
`uname -s`
Darwin
`sysctl kern.maxfiles`
kern.maxfiles: 10240
`sysctl kern.maxfilesperproc`
kern.maxfilesperproc: 10240
`ulimit -n`
7168
`lsof | wc -l`
   10641
`lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30`
  78 suggestd
  80 gitstatus
  80 zsh
  81 Google
  81 Google
  81 zsh
  82 Freedom
  82 Google
  85 Dropzone
  85 Marta
  86 Finder
  91 System
 105 Notificat
 112 Discord
 113 Docker
 124 CoreSpotl
 134 SetappAge
 136 figma_age
 139 com.docke
 140 iTerm2
 146 Google
 156 Google
 163 Google
 169 Discord
 174 corespotl
 178 Electron
 269 UserEvent
 284 Code\x20H
 338 Telegram
 412 Google
```

**Comment by smohamedjavid** [comment]
> @siddarthkay, @flexsurfer, what do you think of adding node_modules to `ignore_dirs` in .`watchmanconfig`? This might fix the issue and improve the overall performance as the amount of files in this folder is huge.

We can try and see.

**Comment by rasom** [comment]
> what do you think of adding node_modules to ignore_dirs in .watchmanconfig

that fixes the issue for me locally

**Comment by codemaster115** [comment]
> what do you think of adding node_modules to ignore_dirs in .watchmanconfig

It works for me locally.

**Comment by rasom** [comment]
oh well, not really
then it goes
```
error: Error: Unable to resolve module node-libs-react-native/globals from /Users/romanvolosovskyi/clj/status-react/index.js: node-libs-react-native/globals could not be found within the project or in these directories:
  node_modules
> 1 | import 'node-libs-react-native/globals';
    |         ^
  2 | import './result/index.js';
  3 |
...
```

**Comment by vkjr** [comment]
Adding `node_modules` to `ignore_dirs` didn't work for me:

<details>

  <summary>make run-metro output</summary>
  
```
To reload the app press "r"
To open developer menu press "d"

jest-haste-map: Watchman crawl failed. Retrying once with node crawler.
  Usually this happens when watchman isn't running. Create an empty `.watchmanconfig` file in your project's root folder or initialize a git or hg repository in your project.
  Error: Watchman error: A non-recoverable condition has triggered.  Watchman needs your help!
The triggering condition was at timestamp=1687360117: opendir(/Users/vkjr/work/projects/status/tmp/status-mobile/node_modules/get-value) -> Too many open files in system
All requests will continue to fail with this message until you resolve
the underlying problem.  You will find more information on fixing this at
https://facebook.github.io/watchman/docs/troubleshooting.html#poison-opendir. Make sure watchman is running for this project. See https://facebook.github.io/watchman/docs/troubleshooting.
Failed to construct transformer:  TypeError: child.send is not a function
    at ChildProcessWorker.initialize (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/workers/ChildProcessWorker.js:173:11)
    at new ChildProcessWorker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/workers/ChildProcessWorker.js:124:10)
    at WorkerPool.createWorker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/WorkerPool.js:44:12)
    at new BaseWorkerPool (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/base/BaseWorkerPool.js:127:27)
    at new WorkerPool (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/WorkerPool.js:30:1)
    at new Worker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/index.js:167:26)
    at HasteMap._getWorker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-haste-map/build/index.js:862:24)
    at HasteMap._processFile (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-haste-map/build/index.js:756:17)
    at HasteMap._buildHasteMap (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-haste-map/build/index.js:809:28)
    at /Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-haste-map/build/index.js:459:33
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at target._send (node:internal/child_process:865:20)
    at target.send (node:internal/child_process:738:19)
    at ChildProcessWorker.initialize (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/workers/ChildProcessWorker.js:173:11)
    at new ChildProcessWorker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/workers/ChildProcessWorker.js:124:10)
    at WorkerPool.createWorker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/WorkerPool.js:44:12)
    at new BaseWorkerPool (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/base/BaseWorkerPool.js:127:27)
    at new WorkerPool (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/WorkerPool.js:30:1)
    at new Worker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-worker/build/index.js:167:26)
    at HasteMap._getWorker (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-haste-map/build/index.js:862:24)
    at HasteMap._processFile (/Users/vkjr/work/projects/status/status-mobile/node_modules/metro/node_modules/jest-haste-map/build/index.js:756:17)
Emitted 'error' event on ChildProcess instance at:
    at node:internal/child_process:869:39
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}

Node.js v18.16.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [run-metro] Error 1
```

</details>

**Comment by smohamedjavid** [comment]
> I guess @smohamedjavid could try to git revert --no-commit c9755979 and see if that fixes it for him.

@rasom @vkjr @codemaster115 - can you help me try this?

**Comment by flexsurfer** [comment]
i don't have any issues on macos 
```
[nix-shell:~/Dev/status-react]$ uname -s
Darwin

[nix-shell:~/Dev/status-react]$ sysctl kern.maxfiles
kern.maxfiles: 245760

[nix-shell:~/Dev/status-react]$ sysctl kern.maxfilesperproc
kern.maxfilesperproc: 122880

[nix-shell:~/Dev/status-react]$ ulimit -n
10240

[nix-shell:~/Dev/status-react]$ lsof | wc -l
132833

[nix-shell:~/Dev/status-react]$ lsof -n | cut -f1 -d' ' | uniq -c | sort | tail -30
 
 117835 watchman

```

**Comment by yakimant** [comment]
@vkjr in your case `kern.maxfiles` and `kern.maxfilesperproc` are also smaller, than others have.
Try to change it too:
```
sudo sysctl -w kern.maxfiles=10485760
sudo sysctl -w kern.maxfilesperproc=1048576
```

**Comment by yakimant** [comment]
@flexsurfer, you have increased settings and watchman opened 100k files.

**Comment by flexsurfer** [comment]
> @flexsurfer, you have increased settings and watchman opened 100k files.

yeah probably i did this in the past, i remember something, @jakubgs suggested

**Comment by vkjr** [comment]
> @rasom @vkjr @codemaster115 - can you help me try this?

@smohamedjavid, this did work!

**Comment by yakimant** [comment]
That's weird, because `watchman` documentation suggests, that raiding the limits is not needed anymore on macOS 10.7 and up. Since they switched from `kqueue` to `fsevents`:https://facebook.github.io/watchman/docs/install.html#macos-file-descriptor-limits

**Comment by flexsurfer** [comment]
https://github.com/status-im/status-mobile/blob/d91e67cae7fd85765389ab64b3ada202c815b771/nix/KNOWN_ISSUES.md?plain=1#L22
https://github.com/status-im/status-mobile/blob/d91e67cae7fd85765389ab64b3ada202c815b771/nix/KNOWN_ISSUES.md?plain=1#L76
is this related ? btw i also run `make nix-upgrade` today

**Comment by jakubgs** [comment]
In CI we do raise the files limit on the hosts to be sure Nix won't hit them:
```yaml
- name: Raise open files system limit
  sysctl:
    sysctl_file: '/etc/sysctl.d/88-ansible.conf'
    name: 'fs.file-max'
    value: '1048576'
    state: present
```
https://github.com/status-im/infra-ci/blob/f50058f000e6834a4f12762f27e29821fdf3df98/ansible/roles/nix-setup/tasks/limits.yml#L2-L7

**Comment by jakubgs** [comment]
@flexsurfer yes, it is related. I should have removed that comment.

**Comment by yakimant** [comment]
`ulimit` changes are not needed for macOS, `sysctl` is enough (according to watchman docs).

**Comment by yakimant** [comment]
So current workarounds to test:

# Workaround 1:
Add `node_modules` to `ignore_dirs` in `.watchmanconfig`

# Workaround 2 (works until reboot):
```
sudo sysctl -w kern.maxfiles=10485760
sudo sysctl -w kern.maxfilesperproc=1048576
```

**Comment by jakubgs** [comment]
If you are getting these errors:
```
mkdir: /nix/var/nix/gcroots/per-user/joesmith: Permission denied
```
Then that's a separate issue tracked here:

* https://github.com/status-im/status-mobile/issues/16344

And you can fix that for now just using this:
```sh
sudo mkdir "/nix/var/nix/profiles/per-user/${USER}" "/nix/var/nix/gcroots/per-user/${USER}"
sudo chown "${USER}" "/nix/var/nix/profiles/per-user/${USER}" "/nix/var/nix/gcroots/per-user/${USER}"
```

**Comment by vkjr** [comment]
> ```
> sudo sysctl -w kern.maxfiles=10485760
> sudo sysctl -w kern.maxfilesperproc=1048576
> ```

@yakimant, that didn't work for me (I also fixed the value in the second line to be also 10485760)

**Comment by mmilad75** [comment]
> sudo sysctl -w kern.maxfiles=10485760
> sudo sysctl -w kern.maxfilesperproc=1048576

Working for me

**Comment by codemaster115** [comment]
> sudo sysctl -w kern.maxfiles=10485760
> sudo sysctl -w kern.maxfilesperproc=1048576

It's working for me.

**Comment by smohamedjavid** [comment]
> sudo sysctl -w kern.maxfiles=10485760
sudo sysctl -w kern.maxfilesperproc=1048576

Increasing the max files works for me.

**Comment by vkjr** [comment]
After cloning `status-mobile` from scratch, restart and applying those lines it works for me too 🥳

**Comment by yakimant** [comment]
If `sysctl` workaround helps - here is how to make it persistent (kept after the reboot):
https://superuser.com/a/1646927/481249

**Comment by smohamedjavid** [comment]
I guess increasing the max files works. 

@yakimant - Is there a way we can increase the limit from Nix when devs run `make run-metro`? It's the same as how Watchman does.

**Comment by flexsurfer** [comment]
do we know if this happened because we added more npm libraries?

**Comment by smohamedjavid** [comment]
> do we know if this happened because we added more npm libraries?

@flexsurfer - I highly doubt it. The npm packages were the same before the Nix upgrade.

**Comment by smohamedjavid** [comment]
@yakimant - Can we try updating the Watchman version inside Nix and see if it resolves?

**Comment by yakimant** [comment]
@smohamedjavid, I will add it to the list. Right now I want to debug and compare with before upgrade to figure out, what caused the issue. I was able to reproduce it locally.

But I guess it's not urgent anymore, let's have `sysctl` workaround, it should not make things worse.
Although having too many files open could cause performance issues potentially.

**Comment by jakubgs** [comment]
Seems to me like the commit that adds support for `fsevents` is extremely old:
https://github.com/facebook/watchman/commit/e0762edf330608b646c29199593ee246fff596ae

And was already included in `4.9.0` version that we used before the `nixpkgs` upgrade.
But it's still trying a newer version is worth a try, at least to know if it makes a difference.

Though we should probably first just make a PR that uses `ignore_vcs` and `ignore_dirs` to limit number of files watched.
Then later we can investigate more deeply and figure out if we can get rid of Watchman entirely.

**Comment by flexsurfer** [comment]
ok now i have this issues as well, it happened only after `make run-android` so its somehow related to android build

**Comment by cammellos** [comment]
> ok now i have this issues as well, it happened only after `make run-android` so its somehow related to android build

that's probably because it's creating a bunch of extra files, and you hit the maximum limit?

**Comment by flexsurfer** [comment]
https://github.com/facebook/watchman/issues/923

**Comment by yakimant** [comment]
During #14944 watchman was upgraded from `4.9.0` (Aug 16, 2017) to `2023.01.30.00`:
https://github.com/status-im/nixpkgs/commit/4e9c02bcc709fe1737a746add0e8e0109133d808

**Comment by yakimant** [comment]
Attmpt to fix the issue by `watchman` downgrade: https://github.com/status-im/status-mobile/pull/16406
